_playwright-tests/UI/CustomRepoCRUD.spec.ts

```diff
@@ -0,0 +1,97 @@
+import { deleteAllRepos } from './helpers/deleteRepositories';
+import { closePopupsIfExist, getRowByNameOrUrl } from './helpers/helpers';
+
+export const repoNamePrefix = 'Repo-CRUD';
+export const randomName = () => (Math.random() + 1).toString(36).substring(2, 6);
+export const repoName = `${repoNamePrefix}-${randomName()}`;
+export const rank = () => Math.floor(Math.random() * 10 + 1).toString();
+export const randomNum = () =>
+  Math.floor(Math.random() * 10 + 1)
+    .toString()
+    .padStart(2, '0');
+
+export const url = `https://stephenw.fedorapeople.org/multirepos/${rank()}/repo${randomNum()}/`;
+
+test.describe('Custom Repositories CRUD', () => {
+  test('Add, Read, update, delete a repo', async ({ page }) => {
+    await test.step('Delete any CRUD test repos that exist', async () => {
+      await navigateToRepositories(page);
+      await closePopupsIfExist(page);
+      await deleteAllRepos(page, `&search=${repoNamePrefix}`);
+    });
+    await test.step('Create a repository', async () => {
+      // Click on the 'Add repositories' button
+      await navigateToRepositories(page);
+      await closePopupsIfExist(page);
+      await page.getByRole('button', { name: 'Add repositories' }).first().click();
+      const repositoryModal = page.locator('div[id^="pf-modal-part"]').first();
+      await expect(repositoryModal).toBeVisible();
+      await expect(page.locator('text="Add custom repositories"')).toBeVisible();
+      // Fill in the repository details
+      await page.getByLabel('Name').fill(`${repoName}`);
+      await page.getByLabel('Introspect only').click();
+      await page.getByLabel('URL').fill(url);
+      await page.getByRole('button', { name: 'Save' }).first().click();
+    });
+    await test.step('Wait for status to be "Valid"', async () => {
+      const row = await getRowByNameOrUrl(page, repoName);
+      await expect(row.getByText('Valid')).toBeVisible({ timeout: 60000 });
+    });
+    await test.step('Read the repo', async () => {
+      // Search for the created repo
+      await page.getByRole('textbox', { name: 'Filter by name/url' }).fill(repoName);
+      await expect(page.getByRole('row', { name: `${repoName}` })).toBeVisible();
+      if (await page.getByLabel('Kebab toggle').first().isDisabled())
+        throw Error("Kebab is disabled when it really shouldn't be");
+      await page.getByLabel('Kebab toggle').first().click();
+      // Click on the Edit button to see the repo
+      await page.getByRole('menuitem', { name: 'Edit' }).click();
+      await expect(page.getByText('Edit custom repository')).toBeVisible();
+      // Assert we can read some values
+      await expect(page.getByTestId('input_name')).toBeVisible();
+      const inputValue = await page.getByTestId('input_name').inputValue();
+      await expect(inputValue).toBe(`${repoName}`);
+      const inputValue2 = await page.getByTestId('input_url').inputValue();
+      expect(inputValue2).toBe(`${url}`);
```

Comment:
Could you use https://playwright.dev/docs/api/class-locatorassertions#locator-assertions-to-have-value ?
Also, why did you decide to go for TestId when in the test before you used `getByLabel`?

Reply:
TestID is more reliable. In some cases, such as the edit modal, it seems to be the only way to use a simple locator because as  `Name` will result in two hits. I can try and use a more complex locator.

Reply:
> as `Name` will result in two hits

Which two hits?

Reply:
There is Name in the edit modal and in the repo table:
`class="pf-v5-c-table__text"`
I am now using byPlaceholder with "exact" match to avoid hitting:
```Name/URL```

See if that works.